seccomp/README.md

```diff
@@ -1,61 +1,225 @@
 
-```
---security-opt seccomp=profile.json
-```
+The following example command starts an interactive container based off the Alpine image and starts a shell process. It also applies the `default.json` seccomp profile to it.
 
-The JSON file is sent from the Docker client to the Docker daemon and the docker daemon
-compiles it into a BPF program using a thin go wrapper around libseccomp: github.com/seccomp/libseccomp-golang.
+   ```
+   $ sudo docker run -it --rm --security-opt seccomp=default.json alpine sh ...
+   ```
 
-The best way to test the effect of your seccomp profiles is to add all capabilities and disable apparmor.
-The following example uses one of the profiles included in this guide to show how to prevent a program from making any system calls at all.
-This is a reliable way to make sure your seccomp profile is enforced.
+The above command sends the JSON file from the client to the daemon where it is compiled into a BPF program using a [thin Go wrapper around libseccomp](https://github.com/seccomp/libseccomp-golang).
 
-```
-$ docker run --rm -it --cap-add ALL --security-opt apparmor=unconfined --security-opt seccomp=seccomp-profiles/deny.json alpine sh
-```
+Docker seccomp profiles operate a whitelist approach that specifies allowed syscalls. All syscalls not on the whitelist are not permitted.
+
+Docker supports many security related technologies. It is possible for other security related technologies to interfere with your testing of seccomp profiles. For this reason, the best way to test the effect of seccomp profiles is to add all *capabilities* and disable *apparmor*. This gives you the confidence that any changes you implement through seccomp are not tainted by *capabilities* and *apparmor*.
+
+The following `docker run` flags add all *capabilities* and disable *apparmor*: `--cap-add ALL --security-opt apparmor=unconfined`.
+
+# <a name="clone"></a>Step 1: Clone the labs GitHub repo
+
+In this step you will clone the lab's GitHub repo so that you have the seccomp profiles that you will use for the remainder of this lab.
+
+1. Clone the labs GitHub repo.
+
+   ```
+   $ git clone https://github.com/riyazdf/dockercon-workshop
+   ```
+
+2. Change into the `dockercon-workshop/seccomp` directory.
+
+   ```
+   $ cd dockercon-workshop/seccomp
+   ```
+
+The remaining steps in this lab will assume that you are running commands from this `dockercon-workshop/seccomp` directory. This will be important when referencing the seccomp profiles on the various `docker run` commands throughout the lab.
+
+# <a name="test"></a>Step 2: Test a seccomp profile
+
+> **NOTE TO LAB OWNER:  THIS STEP DOES NOT CURRENTLY WORK. I'M GUESSING THIS IS BECAUSE THE SECCOMP PROFILE IS APPLIED BEFORE THE CONTAINER IS STARTED AS PER GITHUB ISSUES CITED LATER IN THE LAB GUIDE. I'M LEAVING IT IN FOR THE TIME BEING (NIGELPOULTON@HOTMAIL.COM) BUT AM RAISING THE QUESTION OF WHETHER IT COULD BE REMOVED IN FAVOUR OF STEP 2 WHICH DEMOS SIMILAR CAPABILITIES**
```

Comment:
Well, it sort of works, it just fails in a different way from what you would expect. It fails before the container actually starts because the seccomp policy is applied in runc before the container starts. We need to make a list of all syscalls between applying the profile and starting the container so we can tell people what's the minimum set of system calls necessary to have a container start.